dev/WindowsAppRuntime_BootstrapDLL/MddBootstrap.cpp

```diff
@@ -103,45 +103,38 @@ STDAPI MddBootstrapInitialize(
     }
 
     // Success!
-    ++g_initializationCount;
+    WindowsAppRuntime::MddBootstrap::Activity::Context::Get().SetInitializeData();
 
+    initializationCount = WindowsAppRuntime::MddBootstrap::Activity::Context::Get().GetInitializeData(initializationFrameworkPackageFullName);
     initializeActivity.StopWithResult(
         S_OK,
-        static_cast<UINT32>(g_initializationCount),
-        static_cast<UINT16>(WindowsAppRuntime::MddBootstrap::Activity::Context::Get().GetDDLMFindMethodUsed()),
-        static_cast<PCWSTR>(g_initializationFrameworkPackageFullName.get()),
+        static_cast<UINT32>(initializationCount),
+        static_cast<UINT32>(WindowsAppRuntime::MddBootstrap::Activity::Context::GetIntegrityFlags()),
+        initializationFrameworkPackageFullName,
         static_cast <UINT32>(0),
         static_cast<PCSTR>(nullptr),
         static_cast <unsigned int>(0),
         static_cast<PCWSTR>(nullptr),
         static_cast<PCSTR>(nullptr));
 
-    WindowsAppRuntime::MddBootstrap::Activity::Context::Get().Reset();
-
     return S_OK;
 }
 CATCH_RETURN();
 
 STDAPI_(void) MddBootstrapShutdown() noexcept
 {
-    WindowsAppRuntime::MddBootstrap::Activity::Context::Get().SetMddBootStrapAPI(WindowsAppRuntime::MddBootstrap::Activity::MddBootStrapAPI::Shutdown);
+    PWSTR initializationFrameworkPackageFullName{};
+    auto initializationCount{ WindowsAppRuntime::MddBootstrap::Activity::Context::Get().GetInitializeData(initializationFrameworkPackageFullName) };
+    WindowsAppRuntime::MddBootstrap::Activity::Context::Get().SetMddBootstrapAPI(WindowsAppRuntime::MddBootstrap::Activity::MddBootstrapAPI::Shutdown);
     auto threadCallback = wil::ThreadFailureCallback(wilResultLoggingThreadCallback);
     auto shutdownActivity{
         WindowsAppRuntime::MddBootstrap::Activity::Context::Get().GetShutdownActivity().Start(
-            static_cast<UINT32>(g_initializationCount),
-            g_initializationFrameworkPackageFullName.get()) };
+            static_cast<UINT32>(initializationCount),
+            initializationFrameworkPackageFullName) };
 
     auto lock{ std::lock_guard(g_initializationLock) };
-    const auto initializationCount{ g_initializationCount.load() };
-    if (initializationCount > 1)
+    if (!WindowsAppRuntime::MddBootstrap::Activity::Context::Get().ShouldShutdownNow())
     {
-        // Multiply initialized. Just decrement our count
-        --g_initializationCount;
-        return;
-    }
-    else if (initializationCount == 0)
-    {
-        // Not initialized. Nothing to do
         return;
```

Comment:
This means we only reset state when transitioning count from 1->0 (yay) and only produce telemetry when transitioning count from 1->0 (boo) ?

This skips the reset-state in lines 144-166 and 177-179 AND the shutdownActivity.StopWithResult() on lines 168-175?

The mixed logic doesn't thrill me. Things were already complicated before adding telemetry.

We should report telemetry for every Shutdown call (regardless if count was 1, >1 or 0), and we should avoid mixing up the logic like this. We should...

* decrement the count
* stash data in locals if needed for later use
* reset state if count just transitioned 1->0
* report telemetry
* return